Moroz_Homework_2.cpp

```diff
@@ -0,0 +1,230 @@
+    luckyTicketValue %= divider;
+    divider /= 10;
+  }
+
+  uint16_t secondPart{};
+  for (size_t i = 0; i < 3; i++) {
+    secondPart += (luckyTicketValue / divider);
+    luckyTicketValue %= divider;
+    divider /= 10;
+  }
+
+  if (firstPart != secondPart) {
+    std::cout << "This number is not lucky:(" << std::endl;
+    return;
+  }
+
+  std::cout << "Congratulations! Your number is lucky!" << std::endl;
+}
+
+void reverseNumber() {
+  std::cout << "Please enter number in range "
+            << std::numeric_limits<int32_t>::min() << ".."
+            << std::numeric_limits<int32_t>::max() << std::endl;
+  int32_t numberToReverse{};
+  std::cout << "----->";
+  std::cin >> numberToReverse;
+  if (std::cin.fail()) {
+    std::cin.clear();
+    std::cin.ignore(32768, '\n');
+    std::cout << "Wrong number! Please try again later." << std::endl;
+    return;
+  }
+
+  int64_t reversedNumber{};
+  while (numberToReverse) {
+    reversedNumber += ((numberToReverse % 10));
+    numberToReverse /= 10;
+    if (numberToReverse) {
+      reversedNumber *= 10;
+    }
+  }
+
+  std::cout << "Your reversed number: " << reversedNumber << std::endl;
+}
+
+void sumOfOdd() {
+  constexpr uint16_t numberOfIteration{10};
+  constexpr int16_t minValue{-60};
+  constexpr int16_t maxValue{60};
+  int64_t sumOfOddNumbers{};
+  std::cout << "Please, enter " << numberOfIteration
+            << " numbers to calculate summ of odd numbers:" << std::endl;
+
+  uint16_t counter{};
+  while (counter < numberOfIteration) {
+    int32_t number{};
+    std::cout << "-->";
+    std::cin >> number;
+    if (std::cin.fail()) {
+      std::cin.clear();
+      std::cin.ignore(32768, '\n');
+      std::cout << "Wrong number! Please try again later." << std::endl;
+      return;
+    }
+    if ((number < minValue) || (number > maxValue)) {
+      std::cout << "Invalid value! Break." << std::endl;
+      return;
+    }
+    if ((number % 2) != 0) {
+      sumOfOddNumbers += number;
+    }
+    counter++;
+  }
+
+  std::cout << "The sum of all odd elements is " << sumOfOddNumbers
+            << std::endl;
+}
+
+void bestDivider() {
+  constexpr int16_t minValue{2};
+  std::cout << "Please enter some integer number between " << minValue
+            << " and " << std::numeric_limits<int16_t>::max() << std::endl;
+  int16_t value{};
+
+  std::cin >> value;
+  if (std::cin.fail()) {
+    std::cin.clear();
+    std::cin.ignore(32768, '\n');
+    std::cout << "Wrong number! Please try again later." << std::endl;
+    return;
+  }
+  int16_t dividerValue{value};
+  int16_t maxValue{};
+  while (dividerValue) {
+    if ((value % dividerValue) == 0) {
+      if (dividerValue > 9) {
+        int16_t sumOfDigitsDividerValue{};
+        int16_t temp{dividerValue};
+        while (temp) {
+          sumOfDigitsDividerValue += temp % 10;
+          temp /= 10;
+        }
+        if (sumOfDigitsDividerValue > maxValue) {
+          maxValue = sumOfDigitsDividerValue;
```

Comment:
```suggestion
          maxValue = dividerValue;
```
вы должны не сумму делителя выводить а сам делитель